src/fastfilter.h

```diff
@@ -46,15 +46,15 @@ class CFastFilter
         FastRandomContext insecure_rand;
         vData.resize(FILTER_BYTES);
         // by sampling from a different part of the uint256, we make it harder for an attacker to generate collisions
-        grabFrom = (insecure_rand.rand32() % 7) * 4; // should be 4 byte aligned for speed
+        grabFrom = (1 + (insecure_rand.rand32() % 6)) * 4; // should be 4 byte aligned for speed
```

Comment:
why this change?

Reply:
the current fastfilter is a degenerate bloom filter.  It uses a k (numhashes) == 1.  This is a tradeoff between time and space since the optimum k is (expected entries)/(array size in bits) * ln2.  But it would not be very fast to offer k's up to 8 or 16 by breaking the 32byte hash into 4 byte chunks (and also the 4 byte chunks reversed).

The advantage of ^ is that all the bits of the hash are actually used.  In the current fastfilter case only 32 bits are selected to be used.  If someone grinds these 32 bits to be all 0s we will get entirely conflicts.  Miners do this for block hashes so don't use those bits.  In review, I should probably xor all the bytes of the hash together into a single 32 byte chunk.  This would likely still be quite fast because it will pull from a single cache line.